install.sh

```diff
@@ -304,6 +310,8 @@ build_static=false
 build_release_debug=false
 build_codecoverage=false
 build_address_sanitizer=false
+declare -a cmake_common_options
+declare -a cmake_client_options
```

Comment:
You're declaring these variables as arrays (which is good) and using array append for the --cmake-arg flag (`+=("element"`), but still using string concatenation elsewhere (e.g., `cmake_common_options="${cmake_common_options} element"`). The first string concatenation is going to change the variable type from array to string.

Reply:
Ah, my ignorance of shell scripting is revealed! I'll update the rest of the cmake_common_options additions.